tensorflow/stream_executor/dnn.h

```diff
@@ -766,6 +779,35 @@ class AlgorithmDesc {
   AlgorithmProto proto_;
 };
 
+// Collects parameters for DNN execution plans
+class ExecutionPlanDesc {
```

Comment:
How hard is it to merge this with `AlgorithmDesc` and `AlgorithmConfig`? In general we don't want to explode the number of interfaces. If we can successfully pull this off it's a huge win; it means changes to `core/kernels/...` would be  relatively little, and other users (e.g. XLA) can adopt this much more easily.

Reply:
Done.